src/event/player/PlayerDisplayNameChangeEvent.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\event\player;
+
+use pocketmine\player\Player;
+
+class PlayerDisplayNameChangeEvent extends PlayerEvent{
+
+	/** @var string */
+	private $oldName;
+	/** @var string */
+	private $newName;
+
+	public function __construct(Player $player, string $oldName, string $newName){
+		$this->player = $player;
+		$this->oldName = $oldName;
+		$this->newName = $newName;
+	}
+
+	public function getOldName() : string{
+		return $this->oldName;
+	}
+
+	public function getNewName() : string{
+		return $this->newName;
+	}
+
+	public function setNewName(string $newName) : void{
```

Comment:
I'd prefer this wasn't a thing. It's not clear what the implications of this would be, same as cancellation.
I can approve this much more quickly if the event can't modify the outcome.

Reply:
Yeah true..

Reply:
To be clear, I don't like this because it permits behavioural changes that could break existing plugins, which makes it a much more review-intensive change because we need to figure out whether it makes sense to allow this to be altered (you don't expect your player's name to become Fred when you set it to Steve).

Events that only permit listening to stuff are guaranteed to not have any effect on the code flow and therefore count as pure addition of features. Cancellable and mutable events are much more troublesome.